osis-app/src/main/java/com/scality/osis/healthcheck/S3HealthIndicator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ *Copyright 2021 Scality, Inc.
+ *SPDX-License-Identifier: Apache License 2.0
+ */
+
+package com.scality.osis.healthcheck;
+
+import com.scality.osis.ScalityAppEnv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.net.HttpURLConnection;
+
+@Component("s3")
+public class S3HealthIndicator implements HealthIndicator {
+    private static final Logger logger = LoggerFactory.getLogger(S3HealthIndicator.class);
+
+    @Autowired
+    private ScalityAppEnv appEnv;
+
+    @Override
+    public Health health() {
+        HttpURLConnection connection = null;
+        try {
+            // check if S3 Endpoint is reachable
+            connection = (HttpURLConnection) new java.net.URL(appEnv.getS3Endpoint()).openConnection();
+            connection.setConnectTimeout(appEnv.getS3HealthCheckTimeout());
+            connection.connect();
+        } catch (Exception e) {
```

Comment:
You can be more specific here, and differentiate a timeout from a generic IOException - https://docs.oracle.com/javase/7/docs/api/java/net/URLConnection.html#connect().

This way, you can provide the specific timeout value that was reached in the event of a timeout.

Reply:
does this look any better?
https://github.com/scality/osis/pull/92/commits/1352e1a1c98f04a167eb07e33c5a1e6ffb9a9bf0